bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -1883,6 +1966,79 @@ private static String createCaseDataBaseName(String candidateDbName) {
 		return dbName;
 	}
 
+	/**
+	 * Returns the Examiner object for currently logged in user 
+	 *
+	 * @return A Examiner object.
+	 *
+	 * @throws org.sleuthkit.datamodel.TskCoreException
+	 */
+	public Examiner getCurrentExaminer() throws TskCoreException {
+		
+		String loginName = System.getProperty("user.name");
+		if (loginName == null || loginName.isEmpty()) {
+			throw new TskCoreException("Failed to determine logged in user name.");
+		}
+		
+		CaseDbConnection connection = connections.getConnection();
+		acquireSingleUserCaseReadLock();
+		ResultSet resultSet = null;
+		try {
+			PreparedStatement statement = connection.getPreparedStatement(PREPARED_STATEMENT.SELECT_EXAMINER_BY_LOGIN_NAME);
+			statement.clearParameters();
+			statement.setString(1, loginName);
+			resultSet = connection.executeQuery(statement);
+			if (resultSet.next()) {
+				return new Examiner(resultSet.getLong("examiner_id"), resultSet.getString("login_name"), resultSet.getString("display_name"));
+			}
+			else {
+				throw new TskCoreException("Error getting examaminer for name = " + loginName);
+			}
+					
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error getting examaminer for name = " + loginName, ex);
+		} finally {
+			closeResultSet(resultSet);
+			connection.close();
+			releaseSingleUserCaseReadLock();
+		}
+	}
+	
+	
+	/**
+	 * Returns the Examiner object for given id 
+	 * 
+	 * @param id
+	 * 
+	 * @return Examiner object 
+	 * 
+	 * @throws TskCoreException 
+	 */
+	Examiner getExaminerById(long id) throws TskCoreException {
+		
+		CaseDbConnection connection = connections.getConnection();
+		acquireSingleUserCaseReadLock();
+		ResultSet resultSet = null;
+		try {
+			PreparedStatement statement = connection.getPreparedStatement(PREPARED_STATEMENT.SELECT_EXAMINER_BY_ID);
+			statement.clearParameters();
+			statement.setLong(1, id);
+			resultSet = connection.executeQuery(statement);
+			if (resultSet.next()) {
+				return new Examiner(resultSet.getLong("examiner_id"), resultSet.getString("login_name"), resultSet.getString("full_name"));
+			}
+			else {
+				throw new TskCoreException("Error getting examaminer for id = " + id);
```

Comment:
should this throw an exception or return null/empty optional?

Reply:
I think an Exception is reasonable here, the caller would get the id perhaps from the group_seen table, with foreign key constraint. If we can't find the examiner by id, something is definitely wrong.